arch/xtensa/src/esp32s3/esp32s3_lowputc.c

```diff
@@ -0,0 +1,851 @@
+/****************************************************************************
+ * arch/xtensa/src/esp32s3/esp32s3_lowputc.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/arch.h>
+#include <nuttx/irq.h>
+
+#include <sys/types.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <string.h>
+#include <assert.h>
+#include <errno.h>
+#include <debug.h>
+
+#include <arch/irq.h>
+
+#include "chip.h"
+#include "xtensa.h"
+
+#include "hardware/esp32s3_system.h"
+#include "hardware/esp32s3_uart.h"
+#include "hardware/esp32s3_soc.h"
+
+#include "esp32s3_clockconfig.h"
+#include "esp32s3_config.h"
+#include "esp32s3_gpio.h"
+
+#include "esp32s3_lowputc.h"
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+#ifdef HAVE_UART_DEVICE
+
+#ifdef CONFIG_ESP32S3_UART0
+
+struct esp32s3_uart_s g_uart0_config =
+{
+  .periph = ESP32S3_PERIPH_UART0,
+  .id = 0,
+  .cpuint = -ENOMEM,
```

Comment:
There are situations where `cpuint` is tested to verify the validity of an operation, any error code would work for this however, the function that allocates a CPU interrupt returns `ENOMEM` in case of failure, "Out of Memory" here means "Out of CPU Interrupts".

Reply:
Thank you for explanation